test/find.cpp

```diff
@@ -141,6 +143,66 @@ TEST(FindTests, ByTitle)
     }
 }
 
+TEST(FindTests, ByTitleWithDuplicate)
+{
+  TempZimArchive tza("testZim");
+  zim::writer::Creator creator;
+  creator.startZimCreation(tza.getPath());
+  creator.addItem(std::make_shared<TestItem>("article0", "text/html", "AAA", ""));
+  creator.addItem(std::make_shared<TestItem>("article1", "text/html", "BB", ""));
+  creator.addItem(std::make_shared<TestItem>("article2", "text/html", "BBB", ""));
+  creator.addItem(std::make_shared<TestItem>("article3", "text/html", "BBB", ""));
+  creator.addItem(std::make_shared<TestItem>("article4", "text/html", "BBBB", ""));
+  creator.addItem(std::make_shared<TestItem>("article5", "text/html", "CCC", ""));
+  creator.addItem(std::make_shared<TestItem>("article6", "text/html", "CCC", ""));
+  creator.finishZimCreation();
+
+  zim::Archive archive(tza.getPath());
+  // First binary seach step will look for index 3 (0+6/2) which is a BBB,
+  // but we want to be sure it returns article2 which is the start of the range "BBB*"
+  auto rangeBBB = archive.findByTitle("BBB");
+
+  auto count = 0;
+  for(auto& entry: rangeBBB) {
+    count++;
+    ASSERT_EQ(entry.getTitle().find("BBB"), 0);
+  }
+  ASSERT_EQ(count, 3);
```

Comment:
This piece of code begs to be packaged into a helper macro `CHECK_ENTRY_RANGE()`:

```c++
#define CHECK_ENTRY_RANGE(entry_range, prefix, expected_count) \
  { \
    auto count = 0; \
    for(auto& entry: entry_range) { \
      count++; \
      ASSERT_EQ(entry.getTitle().find(prefix), 0); \
    } \
    ASSERT_EQ(count, expected_count); \
  }
```

Reply:
Done